libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFSDKWebViewStateManager.m

```diff
@@ -82,25 +83,35 @@ + (void)removeUIWebViewCookies:(NSArray *)cookieNames fromDomains:(NSArray *)dom
 
 + (void)removeWKWebViewCookies:(NSArray *)domainNames withCompletion:(nullable void(^)(void))completionBlock {
     NSAssert(domainNames != nil && [domainNames count] > 0, ERR_NO_DOMAIN_NAMES);
-    WKWebsiteDataStore *dateStore = [WKWebsiteDataStore defaultDataStore];
+    WKWebsiteDataStore *dataStore = [WKWebsiteDataStore defaultDataStore];
     NSSet *websiteDataTypes = [NSSet setWithArray:@[ WKWebsiteDataTypeCookies]];
-    [dateStore fetchDataRecordsOfTypes:websiteDataTypes
+    [dataStore fetchDataRecordsOfTypes:websiteDataTypes
                      completionHandler:^(NSArray<WKWebsiteDataRecord *> *records) {
                          NSMutableArray<WKWebsiteDataRecord *> *deletedRecords = [NSMutableArray new];
-                         for ( WKWebsiteDataRecord * record in records) {
+                         for (WKWebsiteDataRecord * record in records) {
+                             // Cookie record display names look like "salesforce.com", "force.com". Make
+                             // them look like proper cookie domain suffixes, for comparison.
+                             NSString *recordDisplayName = [NSString stringWithFormat:@".%@", record.displayName];
                              for(NSString *domainName in domainNames) {
-                                 if ([record.displayName containsString:domainName]) {
+                                 if ([domainName hasSuffix:recordDisplayName]) {
```

Comment:
Since these record type domains are by definition going to be the shortest possible value legitimately representing a cookie domain, I went with `hasSuffix`.